sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/directconnectivity/StoreReader.java

```diff
@@ -248,7 +248,11 @@ public StoreReader(
             for (StoreResult srr : newStoreResults) {
 
                 entity.requestContext.requestChargeTracker.addCharge(srr.requestCharge);
-                BridgeInternal.recordResponse(entity.requestContext.cosmosResponseDiagnostics, entity, srr);
+                try {
+                    BridgeInternal.recordResponse(entity.requestContext.cosmosResponseDiagnostics, entity, srr);
+                } catch (Exception e) {
+                    logger.error("Error occurred while recording response", e);
```

Comment:
Please similar to the original PR, change to "Unexpected failure while recording response"

Reply:
Fixed